components/datetime/src/combo.rs

```diff
@@ -66,232 +69,60 @@ use icu_provider::marker::NeverMarker;
-    type YearStyleOption = (); // no year in a time-only format
-    type TimePrecisionOption = T::TimePrecisionOption;
-    type GluePatternV1Marker = NeverMarker<GluePatternV1<'static>>;
-}
-
-impl<Z> DateTimeNamesMarker for Combo<NeoNeverMarker, NeoNeverMarker, Z>
-where
-    Z: DateTimeNamesMarker,
-{
-    type YearNames = NeverMarker<()>;
-    type MonthNames = NeverMarker<()>;
-    type WeekdayNames = NeverMarker<()>;
-    type DayPeriodNames = NeverMarker<()>;
-    type ZoneEssentials = Z::ZoneEssentials;
-    type ZoneLocations = Z::ZoneLocations;
-    type ZoneGenericLong = Z::ZoneGenericLong;
-    type ZoneGenericShort = Z::ZoneGenericShort;
-    type ZoneSpecificLong = Z::ZoneSpecificLong;
-    type ZoneSpecificShort = Z::ZoneSpecificShort;
-    type MetazoneLookup = Z::MetazoneLookup;
 }
 
-impl<Z> HasConstComponents for Combo<NeoNeverMarker, NeoNeverMarker, Z>
-where
-    Z: HasConstZoneComponent,
-{
-    const COMPONENTS: NeoComponents = NeoComponents::Zone(Z::COMPONENT);
-}
-
-impl<Z> DateTimeMarkers for Combo<NeoNeverMarker, NeoNeverMarker, Z>
-where
-    Z: DateTimeMarkers,
-{
-    type D = NeoNeverMarker;
-    type T = NeoNeverMarker;
-    type Z = Z;
-    type LengthOption = Z::LengthOption; // no date or time: inherit from zone
-    type AlignmentOption = Z::AlignmentOption; // no date or time: inherit from zone
-    type YearStyleOption = (); // no year in a zone-only format
-    type TimePrecisionOption = ();
-    type GluePatternV1Marker = GluePatternV1Marker;
-}
-
-impl<D, T> DateTimeNamesMarker for Combo<D, T, NeoNeverMarker>
-where
-    D: DateTimeNamesMarker,
-    T: DateTimeNamesMarker,
-{
-    type YearNames = D::YearNames;
-    type MonthNames = D::MonthNames;
-    type WeekdayNames = D::WeekdayNames;
-    type DayPeriodNames = T::DayPeriodNames;
-    type ZoneEssentials = NeverMarker<()>;
-    type ZoneLocations = NeverMarker<()>;
-    type ZoneGenericLong = NeverMarker<()>;
-    type ZoneGenericShort = NeverMarker<()>;
-    type ZoneSpecificLong = NeverMarker<()>;
-    type ZoneSpecificShort = NeverMarker<()>;
-    type MetazoneLookup = NeverMarker<()>;
-}
-
-impl<D, T> HasConstComponents for Combo<D, T, NeoNeverMarker>
-where
-    D: HasConstDateComponents,
-    T: HasConstTimeComponents,
-{
-    const COMPONENTS: NeoComponents = NeoComponents::DateTime(D::COMPONENTS, T::COMPONENTS);
-}
+impl<DT, Z> UnstableSealed for Combo<DT, Z> {}
+
+impl<DT, Z> Combo<DT, Z> {
+    /// Gets a mutable reference to the inner date/time field set.
+    ///
+    /// # Examples
+    ///
+    /// Override the length option:
+    ///
+    /// ```no_run
+    /// use icu::datetime::fieldset::{self, Combo};
+    /// use icu::datetime::neo_skeleton::NeoSkeletonLength;
+    ///
+    /// let mut field_set: Combo<fieldset::YMD, fieldset::L> = unimplemented!();
+    ///
+    /// field_set.as_mut_dt().length = NeoSkeletonLength::Medium;
+    /// ```
+    pub fn as_mut_dt(&mut self) -> &mut DT {
```

Comment:
nit (optional): `dt_mut()` or `get_dt_mut()`

thought: also happy with a public field but i get why we want to keep it changeable